### DISCUSSION TITLE
Difference between deleteContact and deleteRecipient

### DISCUSSION THREAD

**Discussion by bublath** [AsamK/signal-cli#862] [Q&A] [answered]
I'm currently enabling these methods for my application via Dbus, but I struggle to understand the actual effect.
Can someone explain the difference, effect and use cases?
Mostly I guess people would use it for cleanup and it seems to me I'd simply would need to call both.

Assumption:
deleteContact just removes things like your own alias name (setContact) and not much more
deleteRecipient removes everything including established trust, stored message data... (so does it supercede deleteContact so I only need to call this?)

**Comment by AsamK** [comment, marked as answer]
That's pretty much it, `deleteContact` just deletes the local contact information like name, color and message expiration time. It can be used to not sync a contact to linked devices anymore.
`deleteRecipient` additionally deletes everything else linked to the contact, like session, sender keys and identity. The only thing not deleted by `deleteRecipient` is group membership information.
After a recipient has been deleted, incoming messages from this recipient will trigger decryption errors, because the session is missing.
